### PR TITLE
*: refine github actions and makefile

### PR DESCRIPTION
**Pull request by xhebox**
Signed-off-by: xhe <xw897002528@gmail.com>

### What problem does this PR solve?

Issue Number: close None

Problem Summary: Some minor improvement.

### What is changed and how it works?

1. ran yaml linters on all yaml files
2. `check-file-perm` is moved into github actions
3. All misc github actions are combined into one workflow
4. removed PR labelers, it needs write permissions, but currently TiDB does not have such permission

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: .github/workflows/integration-test-dumpling.yml [hunk 1]
```diff
@@ -4,55 +4,55 @@ on:
...
   mysql-5735:
-    uses: ./.github/workflows/integration-test-dumpling-common.yml
+    uses: ./.github/integration-test-dumpling-base.yml
     with:
       debug: ${{ github.event_name == 'workflow_dispatch' && github.event.inputs.debug }}
       mysql_version: 5.7.35
 
   mysql-8022:
-    uses: ./.github/workflows/integration-test-dumpling-common.yml
+    uses: ./.github/integration-test-dumpling-base.yml
```
Comment (wuhuizuo) [inline .github/workflows/integration-test-dumpling.yml:49]: it's better to move to `.github/workflow-templates/integration-test-dumpling-base.yaml` ?

Comment (xhebox) [reply .github/workflows/integration-test-dumpling.yml:49]: Sounds good to me. But it looks like the new GHA workflow needs correct permission. I wonder if there is a possibility to make it automatic by tichi.

File: .github/workflows/integration-test-dumpling.yml [hunk 2]
```diff
@@ -36,24 +37,54 @@ on:
...
+        ports:
+          - 3306:3306
+        options: --health-cmd="mysqladmin ping" --health-interval=10s --health-timeout=5s --health-retries=3
+    steps:
+      - name: "checkout repository"
+        uses: actions/checkout@v3
+      - name: "set up golang"
+        uses: actions/setup-go@v3
+        with:
+          go-version: 1.18
```
Comment (hawkingrei) [inline .github/workflows/integration-test-dumpling.yml:68]: We start to use ```go1.19.1```

Comment (xhebox) [reply .github/workflows/integration-test-dumpling.yml]: Modified to use the one in `go.mod`.

**Review by hawkingrei — APPROVED** [review]

**Review by lcwangchao — APPROVED** [review]

**Pull request merged**